webrender/src/renderer.rs

```diff
@@ -1319,14 +1322,18 @@ impl Renderer {
                          ubo_data: &[T],
                          shader: ProgramId,
                          quads_per_item: usize,
-                         color_texture_id: TextureId,
-                         mask_texture_id: TextureId,
+                         textures: &PrimitiveBatchTextures,
                          max_prim_items: usize,
                          projection: &Matrix4D<f32>) {
         self.device.bind_program(shader, &projection);
         self.device.bind_vao(self.quad_vao_id);
-        self.device.bind_texture(TextureSampler::Color, color_texture_id);
-        self.device.bind_texture(TextureSampler::Mask, mask_texture_id);
+
+        // TODO(nical): This will force all unused slots to be bound to the invalid
+        // TextureId. Should we just leave them out?
```

Comment:
FWIW we already explicitly bind to TextureId::invalid in some places such as https://github.com/servo/webrender/blob/74a21e9175a8eb0fc8ce1a18b96c7fb966efb3cb/webrender/src/renderer.rs#L1157
The renderer currently also does that for the color and mask textures every time.

In practice it unbinds the texture since the "invalid" gl handle is 0 under the hood.

I don't know if unbinding an unused texture slot after a pass that used it has any noticeable overhead so I added the comment bu it's probably not a big deal.